dot/core/service.go

```diff
@@ -482,14 +484,22 @@ func (s *Service) DecodeSessionKeys(enc []byte) ([]byte, error) {
 	return rt.DecodeSessionKeys(enc)
 }
 
-// GetRuntimeVersion gets the current RuntimeVersion
-func (s *Service) GetRuntimeVersion(bhash *common.Hash) (
+// GetRuntimeVersion gets the current runtime version.
+func (s *Service) GetRuntimeVersion(blockHash *common.Hash) (
 	version runtime.Version, err error) {
-	rt, err := prepareRuntime(bhash, s.storageState, s.blockState)
+	var blockHashValue common.Hash
+	if blockHash != nil {
+		blockHashValue = *blockHash
+	} else {
+		blockHashValue = s.blockState.BestBlockHash()
+	}
+
+	runtime, err := s.blockState.GetRuntime(blockHashValue)
 	if err != nil {
-		return version, fmt.Errorf("setting up runtime: %w", err)
+		return version, fmt.Errorf("getting runtime: %w", err)
 	}
-	return rt.Version(), nil
+
+	return runtime.Version(), nil
```

Comment:
it would be nice to just cache this, so we don't need to execute the `Version` function every time.  Can also help when we don't store every running runtime instance.

Reply:
It's already cached, see https://github.com/ChainSafe/gossamer/blob/4ea0a703339ca06e617b06eeb7d56308e7b48093/lib/runtime/types.go#L59